Actualización del seed al estado mas reciente. Inclusión de fonts al directorio de producción.

CaroSerrano opened the pull request:
- Agrega cambios que habían quedado pendientes en mi rama respecto al archivo seed, incluyendo funciones auxiliares, nuevas dependencias y variables de entorno.
- Actualiza el script de build para que se agregue la carpeta fonts al directorio de producción. Esto con la esperanza de arreglar el error de Render respecto al archivo de la fuente utilizada para generar el comprobante en PDF.
- Agrega archivos temporales al .gitignore
- Modifica ligeramente la función que devuelve ordenes para que lo haga de manera ordenada.

CaroSerrano committed:
feat: :sparkles: modifica el seed para incorporar productos con imágenes a la tienda del TEST_USER

CaroSerrano committed:
feat: :sparkles: agrega ordenamiento por fecha de creación descendente

CaroSerrano committed:
chore: :see_no_evil: agrega archivos temporales al gitignore

CaroSerrano committed:
Merge branch 'develop' into dev-caro

CaroSerrano committed:
agrega fonts al directorio de producción y variable de entorno al .env.example

niic0o commented:
Hola Caro. Cómo estás? Espero que bien. Gracias por seguir con el proyecto, yo recién martes/miércoles voy a poder revisarte.
1 - podés aceptarte los PR e integrar a producción si crees que está súper checkeado (no recomendado pero yo confío en vos y si algo sale mal después se arregla)
2 - podés seguir haciendo más cambios en tu rama y agregar más PR y después yo reviso.

CaroSerrano commented:
Cómo va Nico? No hay ningún apuro para revisar. Es preferible que no nos salteemos esta etapa para no arrastrar errores innecesariamente. Si sale algún otro cambio, agrego commits a esta PR. Así que tranqui nomas. Espero que estés disfrutando del finde largo! Saluditos

CaroSerrano committed:
fix: cambia el manejo de errores del dashboard

Agrega función handleError al catch del controlador y cambia la clase de errores en el servicio para que se muestren códigos y mensajes apropiados

CaroSerrano committed:
feat: implementa swagger + ts-rest

contratos realizados para auth, products, orders y dashboard

CaroSerrano committed:
feat: agrega contrato para feature user

incluye los endpoints de profile y myData. Agrega handleError al route de profile.

CaroSerrano commented:
Holaa! tarde pero seguro: habemus swagger para los endpoints! se accede desde **/api/docs**. 
Lo implementé usando un sistema de contratos para APIs con la librería ts-rest que nos viene bien porque está preparada para utilizar zod. Más que nada para simplificar la actualización. Vas a ver que cada feature tiene su contrato y luego se importan en el contrato general que está en lib/.
El tema token se maneja automáticamente, es decir, para probar los endpoints protegidos simplemente te logueas primero con el ednpoint login y listo.

**Aclaración**: tuve que setear una versión anterior de React (18.2) para que sea compatible con la librería swagger-ui-react.

niic0o commented:
Hola Caro como estas?? Disculpame que esté atrasado con la revisión, te dije "martes/miercoles" pero más bien será "miercoles/jueves".
Genial que hayas hecho la documentacion con swagger era algo super pendiente que quedó.
Y bueno, eso es todo, vos tranquila sos libre de desarrollar lo que quieras y yo miercoles o jueves me comunico con vos para ver como seguimos y como nos organizamos. Quede con algunas tareas pendientes de otras responsabilidades (universidad) y por eso no desarrollé nada estos días.
